README.md

```diff
@@ -104,6 +104,34 @@ Additional, useful resources:
 
 Note that conventions in the dashboard part of the code base are somewhat different than in the report part.
 
+### Using data
+
+Data is fetched using `api.js`.
+
+If the query parameters of the url do not include values for `offering` and `class`, we will load in fake data from
+the `js/data` folder. This data gets loaded in much the same way as real data, so can be used for testing.
+
+If we do have `offering` and `class` parameters, then `api.js` will first attempt to get the data for the offering and
+class from the portal. To do this it also needs a `token` parameter, which is used to authenticate with the portal and
+expires after one hour. Besides the class and offering data, we will also fetch a firestore JWT from the portal, given
+the classHash (from the fetched class data) and the token. Using this JWT, we can authenticate with Firestore. Once we
+have successfully authenticated, `receivePortalData` is called in `index.ts`, which starts watching the sequence
+structure and answer data.
+
+To test the portal using real data, the easiest way is simply to open a report as a teacher from the portal, and then
+replace the url host and path with `localhost:8080`. Alternatively, if you are able to edit the portal settings for the
+offering, you can add the "Developers Tracked Questions (Local)" report to the External Reports of the offering, and
+when you view the class details, you will see a "Local Tracked Q" button next to "Report" which will link to localhost.
+
+### Additional URL Parameters
+
+Besides the parameters needed for loading data, the report also supports
+
+* `studentId`:  This shows the report for a single student, and removes some UI affordances. The filtering of the student
+                data happens client-side.
+* `iframeQuestionId`: This, combined with a valid `studentId`, will show a stand-alone, full-size iframe containing the
+                model referenced by iframeQuestionId, and the answer saved by studentId (either as state or as a url).
```

Comment:
There is another optional url parameter here that is used by the tests to tell the report to run firestore offline and not make a connection.

Reply:
:+1: